Recruit-Training/Advanced-Recruit-Training/Telemetry-Teaser/src/App.tsx

```diff
@@ -1,26 +1,79 @@
+import { useEffect, useState } from "react";
+
 import BatteryInput from "~/components/batteryInput";
 import Header from "~/components/header";
 import SpeedInput from "~/components/speedInput";
 import WeatherInput from "~/components/weatherInput";
 
 const App = () => {
-  const calculateRange = () => {
-    return;
-  }
+  const [input, setInput] = useState({
+    batteryInput: 0,
+    speedInput: 0,
+    weatherInput: 0,
+  });
+  const [show, setShow] = useState<boolean>(false);
+
+  useEffect(() => {
+    console.log(input);
+  }, [input.speedInput, input.batteryInput, input.weatherInput]);
+
+  const range =
+    -((input.speedInput * input.speedInput * input.batteryInput) / 2500) +
+    4 * input.batteryInput +
+    input.weatherInput;
+  const calculateRange = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    setShow((prev) => !prev);
```

Comment:
do we want to treat setshow as a toggle? do we need a conditional here to see if we should set show to true?